### PR TITLE
[FEATURE] Add docs for configuration module

### PR DESCRIPTION
**Pull request by susannemoog**
See https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/master/Feature-92929-ExtendableConfigurationModule.html
See #1082

### CODE REVIEW

**Review by linawolf — COMMENTED** [review]
I added some comments, suggesting some indexes and

File: Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
+.. include:: /Includes.rst.txt
+
```
Comment (linawolf) [inline Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst:2]: .. index:: 
   pair: Configuration; Module
   TYPO3_CONF_VARS; Validation
   TCA; Validation

File: Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+The configuration module can be found at :guilabel:`System > Configuration`.
+It allows integrators to view and validate the global configuration of TYPO3.
+The module displays all relevant global variables such as
+:php:`TYPO3_CONF_VARS`, :php:`TCA` and many more, in a tree format which is
+easy to browse through. Over time this module got extended to also display
+the configuration of newly introduced features like the middleware stack or
+the event listeners.
+
+
```
Comment (linawolf) [inline Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst:22]: .. index:: 
   Configuration; Module Extension
   Configuration; Module Provider

File: Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+:php:`TYPO3\CMS\Lowlevel\ConfigurationModuleProvider\AbstractProvider` in place
+which already implements the required methods; except :php:`getConfiguration`.
+Please note, when extending this class, the attribute `label` is expected in the
+`__invoke()` method and must therefore be defined in the :file:`Services.yaml`.
+Either a static text or a locallang label can be used.
+
+Since the registration uses the Symfony service container and provides all
+attributes using :php:`__invoke()`, it is even possible to use DI with
+constructor arguments in the provider classes.
+
```
Comment (linawolf) [inline Documentation/ApiOverview/Configuration/ConfigurationModule/Index.rst:117]: .. index:: 
   $GLOBALS; Display
   GlobalVariableProvider

**Review by linawolf — APPROVED** [review]
I would suggest to add some indexes and correct the one heading to Sentence case. Otherwise it looks fine to me

**Comment by marble** [comment]
@linawolf You should edit those files directly and save the changes, which should directly create pull requests for direct integration into this PR.

**Comment by linawolf** [comment]
> @linawolf You should edit those files directly and save the changes, which should directly create pull requests for direct integration into this PR.

I dont seem to be able to do, @marble  that. I guess this is due to the fact that the changes come from susannes fork?

**Comment by marble** [comment]
@linawolf 
> > @linawolf You should edit those files directly and save the changes, which should directly create pull requests for direct integration into this PR.
> 
> I dont seem to be able to do, @marble that. I guess this is due to the fact that the changes come from susannes fork?

① I had seen that Simon Gilli had done it somehow. Let's observe that. ② Now that you have access we can easily do it in a separate branch. ③ If somebody files a pull request s-he has the option to allow change from reviewers to hisher own branch. In that case it's possible too.

**Pull request merged**